base/acme/webapps/acme/WEB-INF/web.xml

```diff
@@ -110,21 +111,30 @@ SPDX-License-Identifier: GPL-2.0-or-later
     </security-constraint>
 
     <security-constraint>
+        <display-name>Allow anybody to get login information</display-name>
+        <web-resource-collection>
+            <url-pattern>/login</url-pattern>
+            <http-method-omission>POST</http-method-omission>
```

Comment:
Why not `<http-method>GET</http-method>` (are we using `DELETE` -- I don't see it -- plus we have a separate `/logout` path)?

Would this make `PUT`/`PATCH` 404 only after authenticating (i.e., 403 -login-> 404)?

My 2c. but I'd rather have them always 404.

Reply:
Let me double-check. IIRC when I tried that I got some kind of warning.

Reply:
Ah I'm not sure how Tomcat `web.xml` config works. I guess I assumed everything not present would 404, which would include endpoints requested with bad methods.

Reply:
OK, I tried specifying just the `<http-method>` for `POST` and `GET` in the `<security-constraint>` for `/login`, and Tomcat generated this warning:
```
SEVERE: For security constraints with URL pattern [/login] only the HTTP methods [POST GET] are covered.
All other methods are uncovered.
```
I think the `<security-constraint>` only cares about the security, not the actual implementation, so we need to include all methods even though we don't support it. The `<http-method-omission>POST</http-method-omission>` basically covers anything other than `POST`, so combined with `<http-method>POST</http-method>` it will cover everything, and there's no more warning.

The actual support is defined using annotations in the service classes. The `ACMELoginService` supports `@POST` and `@GET`. The `POST` authenticates the user, and the `GET` returns the current login info (if any). The `ACMELogoutService` now only supports `@POST`. If a client tries to access using unsupported method, it will get an `HTTP 405 Method Not Allowed`.

Reply:
Ah I understand now. I was thinking this also handled routing for us but I guess not. Thanks for clarifying my understanding!